### ISSUE TITLE
Connected meshes

### ISSUE THREAD

**Issue by KenneyNL** [AlexSTV/Voxelmetric#36]
To be able to create fences and other meshes that connect to each other, have an option that hides/shows parts of a mesh depending on the connection.

In the sample below, the mesh on the left is the actual mesh file. The fences on the right display how parts of the mesh are hidden to create a connected mesh, black is hidden/white is shown.

![Sample](http://i.imgur.com/SqqoYNJ.png)

**Comment by Dalez** [comment]
Wouldn't it be more feasible to change the block to connect to the next fence when another fence is placed down?

E.g. You have a fence, its just a stick currently. You place a fence left, which sets a check to fences x +1, x -1, z +1 and z -1, and if there is a fence on them, update it to be a fence that connects to the way the new fence is.

In the above example, wouldn't you need to create multiple meshes for each "panel" of the fence?

**Comment by Ecu** [comment]
I agree with Dalez, I think having a mesh per connection would be more flexible.  Such as if you want the mesh to be quite different based on how its connecting.  In addition, if random meshing is allowed with it.. you may say have some variation on the mesh based on connections.

**Comment by KenneyNL** [comment]
Agreed too, separated meshes should be better. To my knowledge, there should be 5 meshes made for each connected mesh block: straight, corner, T-split, X-split and stand-alone.

**Comment by Ecu** [comment]
What about things that connect in all six directions?  Such as say BuildCraft (Minecraft mod) pipes?  Probably should account for that as well.
